### PR TITLE
docs: Update Kubernetes docs to use TLS by default

### PR DESCRIPTION
**Pull request by JakeSCahill**
The Helm chart was recently updated to enable TLS by default. This PR adjusts existing docs to account for this change.

### CODE REVIEW

File: docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx [hunk 1]
```diff
@@ -144,7 +152,7 @@ Each Redpanda broker comes with rpk, which is a CLI tool for connecting to and i
 1. Create an alias to simplify the `rpk` commands:
 
   ```bash
-  alias container-rpk="kubectl -n redpanda exec -ti redpanda-0 -c redpanda -- rpk --brokers redpanda-0.redpanda.redpanda.svc.cluster.local.:9093,redpanda-1.redpanda.redpanda.svc.cluster.local.:9093,redpanda-2.redpanda.redpanda.svc.cluster.local.:9093"
+  alias container-rpk="kubectl -n redpanda exec redpanda-0 -c redpanda -- rpk topic --brokers redpanda-0.redpanda.redpanda.svc.cluster.local.:9093,redpanda-1.redpanda.redpanda.svc.cluster.local.:9093,redpanda-2.redpanda.redpanda.svc.cluster.local.:9093 --tls-truststore /etc/tls/certs/default/ca.crt --tls-enabled"
```
Comment (alejandroEsc) [inline docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx:155]:
```suggestion
  alias container-rpk="kubectl -n redpanda exec redpanda-0 -c redpanda -- rpk --brokers redpanda-0.redpanda.redpanda.svc.cluster.local.:9093,redpanda-1.redpanda.redpanda.svc.cluster.local.:9093,redpanda-2.redpanda.redpanda.svc.cluster.local.:9093 --tls-truststore /etc/tls/certs/default/ca.crt --tls-enabled"
```
Not sure if you meant to add the topic to the command, i believe the following commands will fail in that case?

File: docs/deploy/deployment-option/self-hosted/kubernetes/shared/guides/_start-streaming.mdx [hunk 2]
```diff
@@ -3,13 +3,13 @@
 1. Create an alias to simplify the `rpk` commands:
 
   ```bash
-  alias container-rpk="kubectl -n redpanda exec -ti redpanda-0 -c redpanda -- rpk --brokers redpanda-0.redpanda.redpanda.svc.cluster.local.:9093,redpanda-1.redpanda.redpanda.svc.cluster.local.:9093,redpanda-2.redpanda.redpanda.svc.cluster.local.:9093 --tls-truststore /etc/tls/certs/default/ca.crt --tls-enabled --user=redpanda-twitch-account --password=changethispassword --sasl-mechanism SCRAM-SHA-256"
+  alias rpk-topic="kubectl -n redpanda exec -ti redpanda-0 -c redpanda -- rpk topic --brokers redpanda-0.redpanda.redpanda.svc.cluster.local.:9093,redpanda-1.redpanda.redpanda.svc.cluster.local.:9093,redpanda-2.redpanda.redpanda.svc.cluster.local.:9093 --tls-truststore /etc/tls/certs/default/ca.crt --tls-enabled --user=redpanda-twitch-account --password=changethispassword --sasl-mechanism SCRAM-SHA-256"
```
Comment (alejandroEsc) [inline docs/deploy/deployment-option/self-hosted/kubernetes/shared/guides/_start-streaming.mdx:6]: here it makes sense to have the topic as part of the command.

File: docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx (hunk 1 above)
Comment (JakeSCahill) [reply docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx]: thanks @alejandroEsc - fixed 😄

File: docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-deploy.mdx [hunk 3]
```diff
@@ -23,17 +23,13 @@ Make sure that your Kubernetes cluster meets the [requirements](../kubernetes-cl
 
 In this step, you deploy Redpanda and Redpanda Console. Redpanda Console is a developer-friendly web UI for managing and debugging your Redpanda cluster and your applications. Redpanda Console is included as a subchart in the Redpanda Helm chart.
 
-1. Add the chart repository to your Helm installation:
+1. Add the Redpanda Helm chart repository and install cert-manager using Helm:
 
   ```bash
-  helm repo add redpanda https://charts.redpanda.com/
+  helm repo add redpanda https://charts.redpanda.com && helm repo add jetstack https://charts.jetstack.io && helm repo update && helm install cert-manager jetstack/cert-manager  --set installCRDs=true --namespace cert-manager  --create-namespace
```
Comment (joejulian) [inline docs/deploy/deployment-option/self-hosted/kubernetes/kubernetes-deploy.mdx:29]: This isn't very readable like this. I'd prefer reading it like so. It's just as easy to copy/paste.
```suggestion
  helm repo add redpanda https://charts.redpanda.com
  helm repo add jetstack https://charts.jetstack.io
  helm repo update
  helm install cert-manager jetstack/cert-manager  --set installCRDs=true --namespace cert-manager  --create-namespace
```

File: docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx [hunk 4]
```diff
@@ -310,7 +318,7 @@ IP addresses can change. If the IP addresses of your worker nodes change, you mu
 1. Describe the topic:
 
   ```bash
-  rpk topic describe twitch_chat --tls-enabled --tls-truststore=ca.crt --user=redpanda-twitch-account --password=changethispassword --sasl-mechanism SCRAM-SHA-256
+  rpk topic describe twitch_chat --tls-enabled --tls-truststore=ca.crt
```
Comment (joejulian) [inline docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx:321]: Did you want to use your alias here?

File: docs/manage/kubernetes/security/sasl-kubernetes.mdx [hunk 5]
```diff
@@ -166,73 +166,182 @@ When you have SASL enabled for your Redpanda cluster and you have at least one s
...
   :::
 
 1. Grant the new user permissions for a specific topic. The following command grants `describe` privileges to a topic called `myfirsttopic`:
 
+  <Tabs groupId="tls" queryString>
+  <TabItem value="enabled" label="TLS Enabled">
+
   ```bash
   kubectl exec -n redpanda -c redpanda redpanda-0 -- \
-    rpk acl create --allow-principal User:myuser \
```
Comment (joejulian) [inline docs/manage/kubernetes/security/sasl-kubernetes.mdx:203]: The reason many docs sites use words like `myuser`, `mycluster`, etc. It that it makes it very clear to the reader that it's not a fixed keyword. `user1` may be misunderstood to mean that it's an offset in the user table and that we're altering the first user.

File: versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx [hunk 6]
```diff
@@ -310,7 +318,7 @@ IP addresses can change. If the IP addresses of your worker nodes change, you mu
 1. Describe the topic:
 
   ```bash
-  rpk topic describe twitch_chat --tls-enabled --tls-truststore=ca.crt --user=redpanda-twitch-account --password=changethispassword --sasl-mechanism SCRAM-SHA-256
+  rpk topic describe twitch_chat --tls-enabled --tls-truststore=ca.crt
```
Comment (joejulian) [inline versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx:321]: did you want to use the rpk-topic alias here?

**Comment by joejulian** [comment]
If you'd like to add a section on connecting from outside the cluster (all the current examples are running inside with kubectl exec), there's some steps in here that would help: https://docs.google.com/document/d/1FX8eOCvlqT3pO0m_lEfWJM2-4lbJatlV2pS9siSOq8c/edit#heading=h.ahheeeyxlpgn

**Review by joejulian — APPROVED** [review]
Added a few suggestions, but overall it lgtm.

File: docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx (hunk 4 above)
Comment (JakeSCahill) [inline docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx:321]:
```suggestion
  container-rpk describe twitch_chat
```

File: versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx (hunk 6 above)
Comment (JakeSCahill) [reply versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx:321]: this is external. The alias is for `kubectl exec`

File: docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx (hunk 4 above)
Comment (JakeSCahill) [reply docs/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx:321]: this is external - the `rpk-topic` is an alias for `kubectl exec`

File: versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/kubernetes-deploy.mdx [hunk 7]
```diff
@@ -23,17 +23,13 @@ Make sure that your Kubernetes cluster meets the [requirements](../kubernetes-cl
 
 In this step, you deploy Redpanda and Redpanda Console. Redpanda Console is a developer-friendly web UI for managing and debugging your Redpanda cluster and your applications. Redpanda Console is included as a subchart in the Redpanda Helm chart.
 
-1. Add the chart repository to your Helm installation:
+1. Add the Redpanda Helm chart repository and install cert-manager using Helm:
 
   ```bash
-  helm repo add redpanda https://charts.redpanda.com/
+  helm repo add redpanda https://charts.redpanda.com && helm repo add jetstack https://charts.jetstack.io && helm repo update && helm install cert-manager jetstack/cert-manager  --set installCRDs=true --namespace cert-manager  --create-namespace
```
Comment (JakeSCahill) [inline versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/kubernetes-deploy.mdx:29]:
```suggestion
    helm repo add redpanda https://charts.redpanda.com
  helm repo add jetstack https://charts.jetstack.io
  helm repo update
  helm install cert-manager jetstack/cert-manager  --set installCRDs=true --namespace cert-manager  --create-namespace
```

**Comment by JakeSCahill** [comment]
@joejulian we do provide examples of connecting from outside: https://github.com/redpanda-data/documentation/pull/1298/files#diff-1d9ce6fcf85e41c78905d2efc4872a74b55300ac052535ac9ea7f06585e5200bR266

File: versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx [hunk 8]
```diff
@@ -107,7 +107,15 @@ In this step, you create one master and three worker nodes (one worker node for
 
 ## Deploy Redpanda and Redpanda Console
 
-In this step, you deploy Redpanda and Redpanda Console. Redpanda Console is included as a subchart in the Redpanda Helm chart.
+In this step, you deploy Redpanda with self-signed TLS certificates. Redpanda Console is included as a subchart in the Redpanda Helm chart.
+
+1. Add the Redpanda Helm chart repository and install cert-manager using Helm:
+
+  ```bash
+  helm repo add redpanda https://charts.redpanda.com && helm repo add jetstack https://charts.jetstack.io && helm repo update && helm install cert-manager jetstack/cert-manager  --set installCRDs=true --namespace cert-manager  --create-namespace
```
Comment (shorgi) [inline versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/local-guide.mdx:115]:
```suggestion
 helm repo add redpanda https://charts.redpanda.com
 helm repo add jetstack https://charts.jetstack.io
 helm repo updatehelm install cert-manager jetstack/cert-manager  --set installCRDs=true --namespace cert-manager  --create-namespace
```

File: versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/shared/_default-components.mdx [hunk 9]
```diff
@@ -139,13 +140,32 @@ redpanda:
...
 | 31092 | 9094 | Kafka API |
 | 31644 | 9644 | Admin API |
 
-To learn more, see [Networking and Connectivity in Kubernetes](../../../../../manage/kubernetes/networking/networking-and-connectivity).
\ No newline at end of file
+To learn more, see [Networking and Connectivity in Kubernetes](../../../../../manage/kubernetes/networking/networking-and-connectivity).
+
+## TLS Certificates
+
+By default, TLS is enabled in the Helm chart.
```
Comment (shorgi) [inline versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/shared/_default-components.mdx:156]:
```suggestion
By default, TLS is enabled in the Redpanda Helm chart. 
```

File: versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/shared/guides/_external-access-intro.mdx [hunk 10]
```diff
@@ -1,4 +1,4 @@
-Redpanda brokers can advertise an externally accessible address that external clients can connect to.
+If you want to connect to the Redpanda cluster with external clients, Redpanda brokers must advertise an externally accessible address that external clients can connect to. External clients are common in Internet of Things (IoT) environments, or if you use external services that do not implement VPC peering in your network.
```
Comment (shorgi) [inline versioned_docs/version-22.3/deploy/deployment-option/self-hosted/kubernetes/shared/guides/_external-access-intro.mdx:1]:
```suggestion
If you want to connect to the Redpanda cluster with external clients, Redpanda brokers must advertise an externally accessible address that external clients can connect to. External clients are common in Internet of Things (IoT) environments or in systems using external services that do not implement VPC peering in your network.
```

**Review by shorgi — APPROVED** [review]
Minor comments, LGTM

**Pull request merged**